File: typescript/pylon/28693/src/activities.ts
import { Context } from '@temporalio/activity';

export async function greet(name: string): Promise<string> {

  Context.current().metricMeter.createGauge('my_custom_metric').set(Context.current().info.attempt);

 if( Context.current().info.attempt < 3){
   throw new Error("Failed to greet")
 }

  return `Hello, ${name}!`;
}

File: go/pyl/30967/helloworld/activities_v1/activities_.go
package activities_v1

import (
	"context"

	"go.temporal.io/sdk/activity"
)

func Activity(ctx context.Context) (string, error) {
	logger := activity.GetLogger(ctx)
	logger.Info("Activity activities_v1")
	return "activities_v1!", nil
}
